src/engine/output-results.ts

```diff
@@ -221,14 +221,40 @@ function isComplete(resource: DocumentationResource): boolean {
   return resource.status === 'completed';
 }
 
+/**
+ * Helper function to determine whether current state should be skipped and pathway execution should advance
+ * Checks for DocumentReference resource with state label and advance
+ * @param currentState current state
+ * @param resources list of patient resources
+ * @return boolean as to whether pathway execution should advance
+ */
+function shouldAdvance(currentState: State, resources: DomainResource[]): boolean {
+  return resources.some(r => {
+    if (r.resourceType !== 'DocumentReference') return false;
+    const content = (r as DocumentReference).content[0].attachment.data;
+    if (content) {
+      const convertedContent = atob(content);
+      return convertedContent.includes(currentState.label) && convertedContent.includes('Advance');
```

Comment:
If there's a specific format that this text is always going to be, can we have this check for that exact format? Only concern is that the label of the state could contain the word "Advance" or "Advanced" and conflict.

Reply:
Oh I had not thought about that concern, I'll update to check for the exact string.